R/use_cmakelists.R

```diff
@@ -0,0 +1,85 @@
+#'
+#'
+#' @export
+use_cmakelists <- function(project = NULL) {
+  writeLines(
+    c(
+      "cmake_minimum_required(VERSION 3.0.0)",
+      paste0("project(", project, " VERSION 0.1.0)"),
+      "set(CMAKE_EXPORT_COMPILE_COMMANDS ON)",
+      "include(CTest)",
+      "enable_testing()",
+      "",
+      "add_subdirectory(src)",
+      "set(CPACK_PROJECT_NAME ${PROJECT_NAME})",
+      "set(CPACK_PROJECT_VERSION ${PROJECT_VERSION})",
+      "include(CPack)",
+      "set(CMAKE_EXPORT_COMPILE_COMMANDS ON)"
+    ), "CMakeLists.txt"
+  )
+
+  if (!fs::dir_exists("src")) fs::dir_create("src")
+
+  writeLines(
+    c(
+      paste0("add_library(", project),
+      "  connection.cpp",
+      "  DbColumn.cpp",
+      "  DbColumnDataSource.cpp",
+      "  DbColumnDataSourceFactory.cpp",
+      "  DbColumnStorage.cpp",
+      "  DbConnection.cpp",
+      "  DbDataFrame.cpp",
+      "  DbResult.cpp",
+      "  encode.cpp",
+      "  encrypt.cpp",
+      "  logging.cpp",
+      "  PqColumnDataSource.cpp",
+      "  PqColumnDataSourceFactory.cpp",
+      "  PqDataFrame.cpp",
+      "  PqResult.cpp",
+      "  PqResultImpl.cpp",
+      "  PqResultSource.cpp",
+      "  PqUtils.cpp",
+      "  RcppExports.cpp",
+      "  result.cpp",
+      "  DbColumnDataSourceFactory.h",
+      "  DbColumnDataSource.h",
+      "  DbColumnDataType.h",
+      "  DbColumn.h",
+      "  DbColumnStorage.h",
+      "  DbConnection.h",
+      "  DbDataFrame.h",
+      "  DbResult.h",
+      "  DbResultImplDecl.h",
+      "  DbResultImpl.h",
+      "  encode.h",
+      "  integer64.h",
+      "  pch.h",
+      "  PqColumnDataSourceFactory.h",
+      "  PqColumnDataSource.h",
+      "  PqDataFrame.h",
+      "  PqResult.h",
+      "  PqResultImpl.h",
+      "  PqResultSource.h",
+      "  PqUtils.h",
+      "  RPostgres-init.c",
+      "  RPostgres_types.h",
```

Comment:
This needs to be taken from the list of files with glob patterns *.c, *.h, *.cpp .